<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Reuben Thomas" />

<meta name="date" content="2025-07-16" />

<title>Introduction to LODopt</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Introduction to LODopt</h1>
<h4 class="author">Reuben Thomas</h4>
<h4 class="date">2025-07-16</h4>



<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>The <code>LODopt</code> package provides tools for analyzing
associations between cell type abundances and experimental conditions in
single-cell RNA-seq data.</p>
</div>
<div id="quick-start" class="section level2">
<h2>Quick Start</h2>
<div id="example-data" class="section level3">
<h3>Example Data</h3>
<p>We simulate data for 30 samples with 25 clusters, where 4 clusters
(c0, c2, c7 and c14) have changed their abundance.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="co"># Create example count matrix</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>  <span class="co"># Create test data</span></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>  <span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>   <span class="co"># No of clusters in single cell dataset</span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>   K<span class="ot">=</span><span class="dv">25</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a>   <span class="co"># No of cells in tissue</span></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a>   depth <span class="ot">=</span> <span class="fl">1e9</span></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a>   <span class="co"># Total number of samples</span></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a>   nsamp <span class="ot">=</span> <span class="dv">30</span></span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a>   alpha <span class="ot">=</span> <span class="dv">10</span><span class="sc">^</span><span class="fu">runif</span>(K, <span class="at">min=</span><span class="fu">log10</span>(<span class="fl">0.5</span>), <span class="at">max =</span> <span class="fu">log10</span>(<span class="dv">10</span>))</span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a>   p <span class="ot">&lt;-</span> dirmult<span class="sc">::</span><span class="fu">rdirichlet</span>(<span class="at">alpha =</span> alpha) <span class="sc">|&gt;</span> <span class="fu">sort</span>()</span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a>   p <span class="ot">&lt;-</span> p[p <span class="sc">&gt;</span> <span class="fl">0.001</span>]</span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a>   p <span class="ot">&lt;-</span> p<span class="sc">/</span><span class="fu">sum</span>(p)</span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a>   size <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="dv">10</span>, <span class="fu">length</span>(p))</span>
<span id="cb1-15"><a href="#cb1-15" tabindex="-1"></a>   change_mean <span class="ot">=</span> <span class="fu">rep</span>(<span class="dv">1</span>, <span class="fu">length</span>(p))</span>
<span id="cb1-16"><a href="#cb1-16" tabindex="-1"></a>   <span class="do">##changed cluster indices</span></span>
<span id="cb1-17"><a href="#cb1-17" tabindex="-1"></a>   change_mean[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">8</span>,<span class="dv">15</span>)] <span class="ot">=</span> <span class="fu">c</span>(<span class="fl">0.2</span>, <span class="dv">2</span>, <span class="fl">0.2</span>, <span class="dv">2</span>)</span>
<span id="cb1-18"><a href="#cb1-18" tabindex="-1"></a>   <span class="co"># Simulate counts</span></span>
<span id="cb1-19"><a href="#cb1-19" tabindex="-1"></a>   counts_res <span class="ot">&lt;-</span> <span class="fu">simulate_cellCounts_fromTissue</span>(<span class="at">props=</span>p,<span class="at">nsamp=</span>nsamp,<span class="at">depth=</span>depth, <span class="at">size =</span> size, <span class="at">change_mean =</span> change_mean)</span></code></pre></div>
</div>
<div id="set-up-the-required-summarizedexperiment-object" class="section level3">
<h3>Set up the required SummarizedExperiment object</h3>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>  counts <span class="ot">&lt;-</span> counts_res<span class="sc">$</span>counts</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>  pheno_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">sampleID =</span> <span class="fu">paste0</span>(<span class="st">&quot;S&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>),</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>                           <span class="at">groupid =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;group0&quot;</span>, <span class="dv">15</span>), <span class="fu">rep</span>(<span class="st">&quot;group1&quot;</span>, <span class="dv">15</span>)))</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a>  pheno_data <span class="sc">%&lt;&gt;%</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;sampleID&quot;</span>)</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a>  <span class="fu">require</span>(SummarizedExperiment)</span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a>  model_formula <span class="ot">&lt;-</span> <span class="st">&quot;groupid&quot;</span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a>  cellcomp_se <span class="ot">&lt;-</span> <span class="fu">SummarizedExperiment</span>(<span class="at">assays =</span> <span class="fu">list</span>(<span class="at">counts=</span>counts),</span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a>                                      <span class="at">colData =</span> pheno_data,</span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a>                                      <span class="at">metadata =</span> <span class="fu">list</span>(<span class="at">modelFormula =</span> model_formula,</span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a>                                                      <span class="at">coef_of_interest_index =</span> <span class="dv">2</span>,</span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a>                                                      <span class="at">reference_levels_of_variables =</span> <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">&quot;groupid&quot;</span>, <span class="st">&quot;group0&quot;</span>)),</span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a>                                                      <span class="at">random_seed =</span> <span class="dv">123456</span>,</span>
<span id="cb2-17"><a href="#cb2-17" tabindex="-1"></a>                                                      <span class="at">unchanged_cluster_indices =</span> <span class="cn">NULL</span>))</span></code></pre></div>
</div>
<div id="running-association-analysis" class="section level3">
<h3>Running Association Analysis</h3>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>cellcomp_res <span class="ot">&lt;-</span> <span class="fu">logodds_optimized_normFactors</span>(cellcomp_se)</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Running iteration no. 1&quot;</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="co">#&gt; boundary (singular) fit: see help(&#39;isSingular&#39;)</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Running iteration no. 2&quot;</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Running iteration no. 3&quot;</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Found stable solution&quot;</span></span></code></pre></div>
</div>
<div id="summarizing-results" class="section level3">
<h3>Summarizing Results</h3>
<p>The estimate column has the log odds ratio estimates for the
cluster-specific association with groupid while the
estimates_significance has the corresponding raw p-values for the
corresponding null hypothesis that the log odds ratio is equal to 0.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">print</span>(cellcomp_res<span class="sc">$</span>res)</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="co">#&gt;    cluster_id    comparison   estimates estimates_significance</span></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="co">#&gt; 1          c0 groupidgroup1 -1.52100359           1.079388e-17</span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="co">#&gt; 2          c1 groupidgroup1 -0.17344082           2.124728e-01</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a><span class="co">#&gt; 3          c2 groupidgroup1  0.79391408           1.975057e-09</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a><span class="co">#&gt; 4          c3 groupidgroup1  0.04588954           7.117915e-01</span></span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a><span class="co">#&gt; 5          c4 groupidgroup1 -0.04211118           6.974807e-01</span></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a><span class="co">#&gt; 6          c5 groupidgroup1 -0.13292183           2.514397e-01</span></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a><span class="co">#&gt; 7          c6 groupidgroup1 -0.02823263           7.608435e-01</span></span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a><span class="co">#&gt; 8          c7 groupidgroup1 -1.75714635           3.002863e-50</span></span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a><span class="co">#&gt; 9          c8 groupidgroup1  0.13039553           1.830506e-01</span></span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a><span class="co">#&gt; 10         c9 groupidgroup1  0.14547388           2.132688e-01</span></span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a><span class="co">#&gt; 11        c10 groupidgroup1 -0.09124305           5.273222e-01</span></span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a><span class="co">#&gt; 12        c11 groupidgroup1  0.09725929           3.803643e-01</span></span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a><span class="co">#&gt; 13        c12 groupidgroup1  0.02971688           8.012406e-01</span></span>
<span id="cb4-16"><a href="#cb4-16" tabindex="-1"></a><span class="co">#&gt; 14        c13 groupidgroup1 -0.07937818           4.649689e-01</span></span>
<span id="cb4-17"><a href="#cb4-17" tabindex="-1"></a><span class="co">#&gt; 15        c14 groupidgroup1  0.68735275           9.847319e-12</span></span>
<span id="cb4-18"><a href="#cb4-18" tabindex="-1"></a><span class="co">#&gt; 16        c15 groupidgroup1 -0.04634329           7.559317e-01</span></span>
<span id="cb4-19"><a href="#cb4-19" tabindex="-1"></a><span class="co">#&gt; 17        c16 groupidgroup1  0.08155154           4.830891e-01</span></span>
<span id="cb4-20"><a href="#cb4-20" tabindex="-1"></a><span class="co">#&gt; 18        c17 groupidgroup1 -0.11924900           3.281300e-01</span></span>
<span id="cb4-21"><a href="#cb4-21" tabindex="-1"></a><span class="co">#&gt; 19        c18 groupidgroup1 -0.14010580           2.215115e-01</span></span>
<span id="cb4-22"><a href="#cb4-22" tabindex="-1"></a><span class="co">#&gt; 20        c19 groupidgroup1  0.01754712           8.798781e-01</span></span>
<span id="cb4-23"><a href="#cb4-23" tabindex="-1"></a><span class="co">#&gt; 21        c20 groupidgroup1  0.09118643           4.688317e-01</span></span>
<span id="cb4-24"><a href="#cb4-24" tabindex="-1"></a><span class="co">#&gt; 22        c21 groupidgroup1  0.06205020           4.739529e-01</span></span>
<span id="cb4-25"><a href="#cb4-25" tabindex="-1"></a><span class="co">#&gt; 23        c22 groupidgroup1  0.10001258           4.572840e-01</span></span>
<span id="cb4-26"><a href="#cb4-26" tabindex="-1"></a><span class="co">#&gt; 24        c23 groupidgroup1 -0.06205290           5.987648e-01</span></span>
<span id="cb4-27"><a href="#cb4-27" tabindex="-1"></a><span class="co">#&gt; 25        c24 groupidgroup1  0.08948168           5.602581e-01</span></span></code></pre></div>
</div>
</div>
<div id="conclusion" class="section level2">
<h2>Conclusion</h2>
<p>The <code>LODopt</code> package provides a simple interface for
testing associations between cell type abundances and experimental
conditions, with support for multiple statistical methods and
comprehensive result visualization.</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
