<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Introduction to LODopt • LODopt</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to LODopt">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">LODopt</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/intro.html">Quick Start Guide</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/gladstone-institutes/LODopt"><span class="fa fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Introduction to LODopt</h1>
                        <h4 data-toc-skip class="author">Reuben
Thomas</h4>
            
            <h4 data-toc-skip class="date">2025-08-11</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/gladstone-institutes/LODopt/blob/HEAD/vignettes/intro.Rmd" class="external-link"><code>vignettes/intro.Rmd</code></a></small>
      <div class="d-none name"><code>intro.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The <code>LODopt</code> package provides tools for analyzing
associations between cell type abundances and experimental conditions in
single-cell RNA-seq data.</p>
<p>First, let’s load the necessary libraries:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/gladstone-institutes/LODopt" class="external-link">LODopt</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'stats::filter' by 'dplyr::filter' when</span></span>
<span><span class="co">#&gt; loading 'LODopt'</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: MatrixGenerics</span></span>
<span><span class="co">#&gt; Loading required package: matrixStats</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'matrixStats'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:dplyr':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     count</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'MatrixGenerics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span></span>
<span><span class="co">#&gt;     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span></span>
<span><span class="co">#&gt;     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span></span>
<span><span class="co">#&gt;     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span></span>
<span><span class="co">#&gt;     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span></span>
<span><span class="co">#&gt;     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span></span>
<span><span class="co">#&gt;     colWeightedMeans, colWeightedMedians, colWeightedSds,</span></span>
<span><span class="co">#&gt;     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span></span>
<span><span class="co">#&gt;     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span></span>
<span><span class="co">#&gt;     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span></span>
<span><span class="co">#&gt;     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span></span>
<span><span class="co">#&gt;     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span></span>
<span><span class="co">#&gt;     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span></span>
<span><span class="co">#&gt;     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span></span>
<span><span class="co">#&gt;     rowWeightedSds, rowWeightedVars</span></span>
<span><span class="co">#&gt; Loading required package: GenomicRanges</span></span>
<span><span class="co">#&gt; Loading required package: stats4</span></span>
<span><span class="co">#&gt; Loading required package: BiocGenerics</span></span>
<span><span class="co">#&gt; Loading required package: generics</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'generics'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:dplyr':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     explain</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,</span></span>
<span><span class="co">#&gt;     setequal, union</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'BiocGenerics'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:dplyr':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     combine</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     IQR, mad, sd, var, xtabs</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">#&gt;     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">#&gt;     get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,</span></span>
<span><span class="co">#&gt;     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span></span>
<span><span class="co">#&gt;     rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,</span></span>
<span><span class="co">#&gt;     unsplit, which.max, which.min</span></span>
<span><span class="co">#&gt; Loading required package: S4Vectors</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'S4Vectors'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:dplyr':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     first, rename</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:utils':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     findMatches</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     expand.grid, I, unname</span></span>
<span><span class="co">#&gt; Loading required package: IRanges</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'IRanges'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:dplyr':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     collapse, desc, slice</span></span>
<span><span class="co">#&gt; Loading required package: GenomeInfoDb</span></span>
<span><span class="co">#&gt; Warning: package 'GenomeInfoDb' was built under R version 4.5.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'GenomicRanges'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:magrittr':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     subtract</span></span>
<span><span class="co">#&gt; Loading required package: Biobase</span></span>
<span><span class="co">#&gt; Welcome to Bioconductor</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     Vignettes contain introductory material; view with</span></span>
<span><span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'Biobase'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:MatrixGenerics':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     rowMedians</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyMissing, rowMedians</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="quick-start">Quick Start<a class="anchor" aria-label="anchor" href="#quick-start"></a>
</h2>
<div class="section level3">
<h3 id="example-data">Example Data<a class="anchor" aria-label="anchor" href="#example-data"></a>
</h3>
<p>We simulate data for 30 samples with 25 clusters, where 4 clusters
(c0, c2, c7 and c14) have changed their abundance.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create example count matrix</span></span>
<span>  <span class="co"># Create test data</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span>   <span class="co"># No of clusters in single cell dataset</span></span>
<span>   <span class="va">K</span><span class="op">=</span><span class="fl">25</span></span>
<span>   <span class="co"># No of cells in tissue</span></span>
<span>   <span class="va">depth</span> <span class="op">=</span> <span class="fl">1e9</span></span>
<span>   <span class="co"># Total number of samples</span></span>
<span>   <span class="va">nsamp</span> <span class="op">=</span> <span class="fl">30</span></span>
<span>   <span class="va">alpha</span> <span class="op">=</span> <span class="fl">10</span><span class="op">^</span><span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span><span class="va">K</span>, min<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.5</span><span class="op">)</span>, max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span><span class="op">)</span></span>
<span>   <span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">dirmult</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/dirmult/man/rdirichlet.html" class="external-link">rdirichlet</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="va">alpha</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="op">)</span></span>
<span>   <span class="va">p</span> <span class="op">&lt;-</span> <span class="va">p</span><span class="op">[</span><span class="va">p</span> <span class="op">&gt;</span> <span class="fl">0.001</span><span class="op">]</span></span>
<span>   <span class="va">p</span> <span class="op">&lt;-</span> <span class="va">p</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></span>
<span>   <span class="va">size</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">)</span></span>
<span>   <span class="va">change_mean</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">)</span></span>
<span>   <span class="co">##changed cluster indices</span></span>
<span>   <span class="va">change_mean</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">3</span>,<span class="fl">8</span>,<span class="fl">15</span><span class="op">)</span><span class="op">]</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.2</span>, <span class="fl">2</span>, <span class="fl">0.2</span>, <span class="fl">2</span><span class="op">)</span></span>
<span>   <span class="co"># Simulate counts</span></span>
<span>   <span class="va">counts_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/simulate_cellCounts_fromTissue.html">simulate_cellCounts_fromTissue</a></span><span class="op">(</span>props<span class="op">=</span><span class="va">p</span>,nsamp<span class="op">=</span><span class="va">nsamp</span>,depth<span class="op">=</span><span class="va">depth</span>, size <span class="op">=</span> <span class="va">size</span>, change_mean <span class="op">=</span> <span class="va">change_mean</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="set-up-the-required-summarizedexperiment-object">Set up the required SummarizedExperiment object<a class="anchor" aria-label="anchor" href="#set-up-the-required-summarizedexperiment-object"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>  <span class="va">counts</span> <span class="op">&lt;-</span> <span class="va">counts_res</span><span class="op">$</span><span class="va">counts</span></span>
<span></span>
<span>  <span class="va">pheno_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>sampleID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"S"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">30</span><span class="op">)</span>,</span>
<span>                           groupid <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"group0"</span>, <span class="fl">15</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"group1"</span>, <span class="fl">15</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">pheno_data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/compound.html" class="external-link">%&lt;&gt;%</a></span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"sampleID"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span>
<span>  <span class="va">model_formula</span> <span class="op">&lt;-</span> <span class="st">"groupid"</span></span>
<span>  <span class="va">cellcomp_se</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">SummarizedExperiment</a></span><span class="op">(</span>assays <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>counts<span class="op">=</span><span class="va">counts</span><span class="op">)</span>,</span>
<span>                                      colData <span class="op">=</span> <span class="va">pheno_data</span>,</span>
<span>                                      metadata <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>modelFormula <span class="op">=</span> <span class="va">model_formula</span>,</span>
<span>                                                      coef_of_interest_index <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                                                      reference_levels_of_variables <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"groupid"</span>, <span class="st">"group0"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                                                      random_seed <span class="op">=</span> <span class="fl">123456</span>,</span>
<span>                                                      unchanged_cluster_indices <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="running-association-analysis">Running Association Analysis<a class="anchor" aria-label="anchor" href="#running-association-analysis"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cellcomp_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/logodds_optimized_normFactors.html">logodds_optimized_normFactors</a></span><span class="op">(</span><span class="va">cellcomp_se</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Running iteration no. 1"</span></span>
<span><span class="co">#&gt; boundary (singular) fit: see help('isSingular')</span></span>
<span><span class="co">#&gt; [1] "Running iteration no. 2"</span></span>
<span><span class="co">#&gt; [1] "Running iteration no. 3"</span></span>
<span><span class="co">#&gt; [1] "Found stable solution"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="summarizing-results">Summarizing Results<a class="anchor" aria-label="anchor" href="#summarizing-results"></a>
</h3>
<p>The estimate column has the log odds ratio estimates for the
cluster-specific association with groupid while the
estimates_significance has the corresponding raw p-values for the
corresponding null hypothesis that the log odds ratio is equal to 0.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">cellcomp_res</span><span class="op">$</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt;    cluster_id    comparison   estimates estimates_significance</span></span>
<span><span class="co">#&gt; 1          c0 groupidgroup1 -1.52100359           1.079388e-17</span></span>
<span><span class="co">#&gt; 2          c1 groupidgroup1 -0.17344082           2.124728e-01</span></span>
<span><span class="co">#&gt; 3          c2 groupidgroup1  0.79391408           1.975057e-09</span></span>
<span><span class="co">#&gt; 4          c3 groupidgroup1  0.04588954           7.117915e-01</span></span>
<span><span class="co">#&gt; 5          c4 groupidgroup1 -0.04211118           6.974807e-01</span></span>
<span><span class="co">#&gt; 6          c5 groupidgroup1 -0.13292183           2.514397e-01</span></span>
<span><span class="co">#&gt; 7          c6 groupidgroup1 -0.02823263           7.608435e-01</span></span>
<span><span class="co">#&gt; 8          c7 groupidgroup1 -1.75714635           3.002863e-50</span></span>
<span><span class="co">#&gt; 9          c8 groupidgroup1  0.13039553           1.830506e-01</span></span>
<span><span class="co">#&gt; 10         c9 groupidgroup1  0.14547388           2.132688e-01</span></span>
<span><span class="co">#&gt; 11        c10 groupidgroup1 -0.09124305           5.273222e-01</span></span>
<span><span class="co">#&gt; 12        c11 groupidgroup1  0.09725929           3.803643e-01</span></span>
<span><span class="co">#&gt; 13        c12 groupidgroup1  0.02971688           8.012406e-01</span></span>
<span><span class="co">#&gt; 14        c13 groupidgroup1 -0.07937818           4.649689e-01</span></span>
<span><span class="co">#&gt; 15        c14 groupidgroup1  0.68735275           9.847319e-12</span></span>
<span><span class="co">#&gt; 16        c15 groupidgroup1 -0.04634329           7.559317e-01</span></span>
<span><span class="co">#&gt; 17        c16 groupidgroup1  0.08155154           4.830891e-01</span></span>
<span><span class="co">#&gt; 18        c17 groupidgroup1 -0.11924900           3.281300e-01</span></span>
<span><span class="co">#&gt; 19        c18 groupidgroup1 -0.14010580           2.215115e-01</span></span>
<span><span class="co">#&gt; 20        c19 groupidgroup1  0.01754712           8.798781e-01</span></span>
<span><span class="co">#&gt; 21        c20 groupidgroup1  0.09118643           4.688317e-01</span></span>
<span><span class="co">#&gt; 22        c21 groupidgroup1  0.06205020           4.739529e-01</span></span>
<span><span class="co">#&gt; 23        c22 groupidgroup1  0.10001258           4.572840e-01</span></span>
<span><span class="co">#&gt; 24        c23 groupidgroup1 -0.06205290           5.987648e-01</span></span>
<span><span class="co">#&gt; 25        c24 groupidgroup1  0.08948168           5.602581e-01</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<p>The <code>LODopt</code> package provides a simple interface for
testing associations between cell type abundances and experimental
conditions, with support for multiple statistical methods and
comprehensive result visualization.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Reuben Thomas, Ayushi Agrawal, Natalie Elphick and Michela Traglia.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with pkgdown 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
